src/lib/ui/thermostat/ClimateSettingStatus.tsx

```diff
@@ -0,0 +1,76 @@
+import type { ClimateSetting } from 'seamapi'
+
+import { ThermostatCoolIcon } from 'lib/icons/ThermostatCool.js'
+import { ThermostatHeatIcon } from 'lib/icons/ThermostatHeat.js'
+import { ThermostatHeatCoolIcon } from 'lib/icons/ThermostatHeatCool.js'
+import { ThermostatOffIcon } from 'lib/icons/ThermostatOff.js'
+
+interface ClimateSettingStatusProps {
+  setting: ClimateSetting
+  iconPlacement?: 'left' | 'right'
+}
+
+export function ClimateSettingStatus({
+  setting,
+  iconPlacement = 'left',
+}: ClimateSettingStatusProps): JSX.Element {
+  return (
+    <div className='seam-climate-setting-status'>
+      {iconPlacement === 'left' && (
+        <ClimateSettingIcon mode={setting.hvac_mode_setting} />
+      )}
+      <Content
+        mode={setting.hvac_mode_setting}
+        coolingSetPoint={setting.cooling_set_point_fahrenheit}
+        heatingSetPoint={setting.heating_set_point_fahrenheit}
+      />
+      {iconPlacement === 'right' && (
+        <ClimateSettingIcon mode={setting.hvac_mode_setting} />
+      )}
+    </div>
+  )
+}
+
+function ClimateSettingIcon(props: {
+  mode: ClimateSetting['hvac_mode_setting']
+}): JSX.Element | null {
+  const { mode } = props
+
+  return (
+    <div className='seam-climate-setting-status-icon'>
+      {mode === 'cool' && <ThermostatCoolIcon />}
+      {mode === 'heat' && <ThermostatHeatIcon />}
+      {mode === 'heatcool' && <ThermostatHeatCoolIcon />}
+      {mode === 'off' && <ThermostatOffIcon />}
+    </div>
+  )
+}
+
+function Content(props: {
+  mode: ClimateSetting['hvac_mode_setting']
+  coolingSetPoint: ClimateSetting['cooling_set_point_fahrenheit']
+  heatingSetPoint: ClimateSetting['cooling_set_point_fahrenheit']
+}): JSX.Element | null {
+  const { mode, coolingSetPoint, heatingSetPoint } = props
+  const hasCoolingSetPoint = coolingSetPoint !== undefined
+  const hasHeatingSetPoint = heatingSetPoint !== undefined
```

Comment:
```suggestion
  const hasCoolingSetPoint = coolingSetPoint != null
  const hasHeatingSetPoint = heatingSetPoint != null
```

Reply:
The type we have for it has them as possibly undefined

```
type ClimateSetting = {
    automatic_heating_enabled?: boolean | undefined;
    automatic_cooling_enabled?: boolean | undefined;
    hvac_mode_setting?: "off" | "heat" | "cool" | "heatcool" | undefined;
    cooling_set_point_celsius?: number | undefined;
    heating_set_point_celsius?: number | undefined;
    cooling_set_point_fahrenheit?: number | undefined;
    heating_set_point_fahrenheit?: number | undefined;
    manual_override_allowed: boolean;
}
```

Reply:
Interesting, our general convention is to return optional properties but not return them as existing and undefined. This may be an upstream issues with the types.